src/openfl/geom/Matrix3D.hx

```diff
@@ -765,9 +765,15 @@ class Matrix3D {
 	
 	public function prependTranslation (x:Float, y:Float, z:Float):Void {
 		
-		var m = new Matrix3D ();
-		m.position = new Vector3D (x, y, z);
-		this.prepend (m);
+		var m11 = rawData[0], m21 = rawData[4], m31 = rawData[8],  m41 = rawData[12];
+		var m12 = rawData[1], m22 = rawData[5], m32 = rawData[9],  m42 = rawData[13];
+		var m13 = rawData[2], m23 = rawData[6], m33 = rawData[10], m43 = rawData[14];
+		var m14 = rawData[3], m24 = rawData[7], m34 = rawData[11], m44 = rawData[15];
+		
+		rawData[12] = x * m11 + y * m21 + z * m31 + m41;
+		rawData[13] = x * m12 + y * m22 + z * m32 + m42;
+		rawData[14] = x * m13 + y * m23 + z * m33 + m43;
+		rawData[15] = x * m14 + y * m24 + z * m34 + m44;
```

Comment:
Just also do the `prependScale`as well now 😛

Reply:
Done, please check.